### ISSUE TITLE
Copy of skeleton not linked

### ISSUE THREAD

**Issue by zozorg** [synfig/synfig#453]
**Synfig version & platform**:
linux 1.3.4

**Issue description**:
When copying a skeleton and its linked splines, by copy/pasting or embedding an imported canvas, the resulting skeleton isn't linked to the splines.
I'm willing to post a bounty when confirmed, I assume this one must be somewhat hard to fix. What $$$ would sound reasonable?

**Comment by morevnaproject** [comment]
> Just create a skeleton with 3 bones, duplicate it, and move the bones around.

Yes, I see what you mean. When I duplicate skeleton layer (within the same document) it creates copy of each bone. But those copies still reference parents from old skeleton.

**Comment by morevnaproject** [comment]
In general any copy operations related with bones are dangerous at the moment. For example this one - #455.

**Comment by zozorg** [comment]
Posted a [$50 bounty](https://www.bountysource.com/issues/51184738-cannot-copy-a-skeleton)

**Comment by zozorg** [comment]
+10$ bounty, total $60

**Comment by src-r-r** [comment]
I'm taking a look at this issue now.

**Comment by src-r-r** [comment]
I've been looking through the source. I think part of the problem is that the design of the bone/skeleton relationship is not very structured. Bones are just there: they don't belong to anything. Is this by intention?

Taking a cue from Blender, I propose skeletons take on a more active role, so that a user can select a skeleton, double click, and enter the skeleton group to modify the bones.

Does this work for the design?

**Comment by rodolforg** [comment]
@src-r-r The possible cause of this problem is a wrong duplication of the ValueNode that build the relationships among bones themselves.
Maybe LayerSkeleton should override Layer::clone() method to fix the bones' parenting after duplication.

**Comment by locky504** [comment]
I'm working on this issue. PR #3702 submitted.
